contrib/guix/manifest.scm

```diff
@@ -0,0 +1,155 @@
+         ;; 3. Build a cross-compiled libc with XGCC-SANS-LIBC and XKERNEL,
+         ;; derived from BASE-LIBC
+         (xlibc (cross-libc target
+                            base-libc
+                            xgcc-sans-libc
+                            xbinutils
+                            xkernel))
+         ;; 4. Build a cross-compiling gcc targeting XLIBC, derived from
+         ;; BASE-GCC
+         (xgcc (cross-gcc target
+                          #:xgcc base-gcc
+                          #:xbinutils xbinutils
+                          #:libc xlibc)))
+    ;; Define a meta-package that propagates the resulting XBINUTILS, XLIBC, and
+    ;; XGCC
+    (package
+      (name (string-append target "-toolchain"))
+      (version (package-version xgcc))
+      (source #f)
+      (build-system trivial-build-system)
+      (arguments '(#:builder (begin (mkdir %output) #t)))
+      (propagated-inputs
+       `(("binutils" ,xbinutils)
+         ("libc" ,xlibc)
+         ("libc:static" ,xlibc "static")
+         ("gcc" ,xgcc)))
+      (synopsis (string-append "Complete GCC tool chain for " target))
+      (description (string-append "This package provides a complete GCC tool
+chain for " target " development."))
+      (home-page (package-home-page xgcc))
+      (license (package-license xgcc)))))
+
+(define* (make-bitcoin-cross-toolchain target
+                                  #:optional
+                                  (base-gcc-for-libc gcc-5)
+                                  (base-kernel-headers linux-libre-headers-4.19)
+                                  (base-libc glibc-2.27)
+                                  (base-gcc (make-gcc-rpath-link
+                                             (make-ssp-fixed-gcc gcc-9))))
+  "Convienience wrapper around MAKE-CROSS-TOOLCHAIN with default values
+desirable for building Bitcoin Core release binaries."
+  (make-cross-toolchain target
+                   base-gcc-for-libc
+                   base-kernel-headers
+                   base-libc
+                   base-gcc))
+
+(packages->manifest
+ (list ;; The Basics
+       bash
+       tcsh
+       which
+       coreutils
+       util-linux
+       ;; File(system) inspection
+       file
+       grep
+       diffutils
+       findutils
+       ;; File transformation
+       patch
+       gawk
+       sed
+       ;; Compression and archiving
+       tar
+       bzip2
+       gzip
+       xz
+       zlib
+       ;; Build tools
+       gnu-make
+       libtool
+       autoconf
+       automake
+       pkg-config
+       ;; Scripting
+       perl
+       python
+       ;; Toolchains
+       (make-gcc-toolchain gcc-9 glibc-2.27)
+       (make-bitcoin-cross-toolchain "riscv64-linux-gnu" gcc-8)
+       (make-bitcoin-cross-toolchain "x86_64-linux-gnu")
+       (make-bitcoin-cross-toolchain "i686-linux-gnu")
+       (make-bitcoin-cross-toolchain "aarch64-linux-gnu")
+       (make-bitcoin-cross-toolchain "arm-linux-gnueabihf" gcc-6)))
```

Comment:
Would be nice to explain why three different versions of gcc are used. Why is it not possible to use just one? Maybe the most popular version of gcc, which might be the one in the latest Ubuntu LTS release (we also use that version for gitian builds)

Reply:
Clarified in cd3e947f50db7cfe05c05b368c25742193729a62